docs/vm-configuration.md

```diff
@@ -0,0 +1,265 @@
+different host, there is a non-negligible chance that changes will be required
+in backend config of the XML configuration. For example, the ports currently
+used for VNC may not be available on the new host, or the storage devices that
+need to be used may require a different /dev/sdNN device path.
+
+There are ways to minimise these differences, to increase portability between
+hosts. For example, instead of hardcoding a VNC port, allow libvirt to perform
+automatic allocation, or instead of using an unstable /dev/sdNN path, Linux
+provides /dev/disk/by-path/XXX symlinks which might be stable across hosts.
+
+
+Security issues
+---------------
+
+There are many security features in libvirt, however, they are primarily focused
+around protection from malicious or misbehaving guests. For example, sVirt will
+stop guests attacking each other, or the host. NWFilter will stop guests doing
+IP/MAC/ARP spoofing on the network.
+
+There is a fine grained access control mechanism for the host facing management
+API, however, the checks are performed against the (subject, object, action)
+tuple. So while you can block a user or application from being able to invoke
+the 'virDomainDefineXML' action, if you allow this API, you cannot then control
+what information is passed in its parameters. For example, when configuring
+disks in the domain XML, arbitrary host paths can be specified as the backing
+storage for the virtual disk. Libvirt will happily let /etc/passwd be specified
+as the disk backing store. When libvirt starts the QEMU instance, it will then
+tell sVirt to grant access to the /etc/passwd file.
+
+IOW, any application which can invoke the virDomainDefineXML method on the
+libvirtd daemon has a privilege level which is equivalent to that of libvirtd
+itself, which is root. Thus applications calling virDomainDefineXML (and many
+other APIs) which accept configuration input from untrusted users, to feed into
+an XML document, must be very careful in their data handling.
+
+
+Config states
+-------------
+
+Naively one might expect a virtual machine to have a single document describing
+its configuration. In practice, this is unreasonably restrictive.
+
+The configuration provided by a tenant or management application will typically
+not express every single frontend or backend setting. When the XML document is
+provided to libvirt, it will fill in any important missing pieces of information
+in an attempt to fully represent the virtual hardware configuration and backend
+integration. For example, if the tenant did not supply PCI addresses, libvirt
+will assign them prior to starting QEMU. Even when libvirt fills in missing
+data this is only ever libvirt's best attempt at a complete virtual hardware
+configuration. For example, for ISA serial ports, libvirt does not record the
+hardware I/O port associated with the serial port, since that is not currently
+a configurable option & thus can be assumed stable without being explicitly
```

Comment:
If you make use of a feature that isn't supported by older QEMU, then yes, you of course can't migrate to older QEMU versions. If you merely record an extra bit of information in the XML file, that old libvirt didn't record, but this newly recorded info does not rely on new QEMU features, then libvirt should let you migrate to older versions.